v1.7.0/backing-image-manager/rockcraft.yaml

```diff
@@ -0,0 +1,144 @@
+
+version: "v1.7.0"
+
+# NOTE(aznashwan): the base for the image is the Suse Linux Enterprise
+# Base Container Image (SLE BCE) Service Pack 6 which ships with Linux 6.4,
+# and is thus most comparable to 24.04:
+# https://github.com/longhorn/backing-image-manager/blob/v1.7.0/package/Dockerfile#L3
+base: ubuntu@24.04
+
+build-base: ubuntu@24.04
+platforms:
+  amd64:
+
+
+services:
+  backing-image-manager:
+    startup: enabled
+    override: replace
+
+    # https://github.com/longhorn/backing-image-manager/blob/v1.7.0/package/Dockerfile#L31-L33
+    command: /tini -- [ backing-image-manager ]
+
+    on-success: shutdown
+    on-failure: shutdown
+
+entrypoint-service: backing-image-manager
+
+parts:
+
+  # NOTE(aznashwan): the binary is built within a Docker container
+  # which is set up by Rancher's Dapper tool: https://github.com/rancher/dapper
+  # The setup steps for the build container are contained within this Dockerfile:
+  # https://github.com/longhorn/backing-image-manager/blob/v1.7.0/Dockerfile.dapper
+  setup-build-env:
+    plugin: nil
+
+    build-packages:
+      # https://github.com/longhorn/backing-image-manager/blob/v1.7.0/Dockerfile.dapper#L20-L24
+      - cmake
+      - wget
+      - curl
+      - git
+      - less
+      - file
+      - libkmod-dev             # libkmod-devel
+      - libnl-3-dev             # libnl3-devel
+      - linux-headers-6.8.0-41  # linux-glibc-devel
+      - pkg-config
+      - psmisc
+      - fuse
+      - librdmacm-dev           # librdmacm1
+      - rdmacm-utils            # librdmacm-utils
+      - libibverbs-dev          # libibverbs
+      - libaio-dev              # libaio-devel
+      - libc6-dev               # glibc-devel, glibc-devel-static
+      - iptables
+      - libltdl7
+      - libdevmapper1.02.1      # libdevmapper1_03
+      - iproute2
+      - jq
+      - gcc
+
+  # NOTE(aznashwan): the Makefile targets are just the scripts found in the
+  # scripts/ directory which are executed within the Dapper build container:
+  # https://github.com/longhorn/backing-image-manager/blob/v1.7.0/Makefile#L1
+  # https://github.com/longhorn/backing-image-manager/blob/v1.7.0/Makefile#L10-L11
+  build-lonhorn-backing-image-manager:
+    plugin: nil
+    after: ["setup-build-env"]
+
+    source-type: git
+    source: https://github.com/longhorn/backing-image-manager
+    source-tag: v1.7.0
+    source-depth: 1
+
+    build-snaps:
+      # https://github.com/longhorn/backing-image-manager/blob/v1.7.0/Dockerfile.dapper#L1
+      - go/1.22/stable
+
+    override-build: |
+      set -eux -o pipefail
+
+      cd $CRAFT_PART_SRC
+
+      bash ./scripts/build
+
+      # https://github.com/longhorn/backing-image-manager/blob/v1.7.0/package/Dockerfile#L23
+      mkdir -p $CRAFT_PART_INSTALL/usr/local/bin
+      cp $CRAFT_PART_SRC/bin/backing-image-manager-${CRAFT_ARCH_BUILD_FOR} $CRAFT_PART_INSTALL/usr/local/bin/backing-image-manager
+
+  # Pulls a pre-built binary release of the `tini` init system.
```

Comment:
We have left `tini` mostly in the spirit of full compatibility with the upstream image.

It is present in the original image mainly for:
- reaping zombie processes
- forwarding POSIX signals to the real service

Afaik Pebble does reap zombies, but afaik it does **not** forward signals (which basically invalidates tini's signal handling features).

There is a chance `tini` does something extra I'm not aware of, so I would personally go with keeping it there just in case, but am open to removing it from the Pebble service definition at least...

> If needed we should look into building from source later on.

FWIW there is an Ubuntu package with `tini`, I could simply switch to copying the `tini` binary from it if if you think that'd help a bit in the short term?

Reply:
Let's make a note of this to stage it from the archive(`tini` package) later on, thanks!